cmd/gitserver/internal/git/gitcli/diff.go

```diff
@@ -41,3 +45,100 @@ func buildRawDiffArgs(base, head api.CommitID, typ git.GitDiffComparisonType, pa
 		"--",
 	}, paths...)
 }
+
+func (g *gitCLIBackend) ChangedFiles(ctx context.Context, base, head string) ([]gitdomain.PathStatus, error) {
+	args := []string{
+		"diff-tree",
+		"-r",
+		"--root",
+		"--format=format:",
+		"--no-prefix",
+		"--name-status",
+		"--no-renames",
+		"-z",
+	}
+
+	headOID, err := g.revParse(ctx, head)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve head commit %q", head)
+	}
+
+	if base != "" {
+		baseOID, err := g.revParse(ctx, base)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to resolve base commit %q", base)
+		}
+
+		args = append(args, string(baseOID))
+	}
+
+	args = append(args, string(headOID))
```

Comment:
super nit: I wonder if this would be slightly easier to read if it was colocated (I read this twice before I realized that we actually print `base head` in this order and not `head base` into the args

```suggestion
	if base != "" {
		baseOID, err := g.revParse(ctx, base)
		if err != nil {
			return nil, errors.Wrapf(err, "failed to resolve base commit %q", base)
		}

		args = append(args, string(baseOID))
	}

	headOID, err := g.revParse(ctx, head)
	if err != nil {
		return nil, errors.Wrapf(err, "failed to resolve head commit %q", head)
	}
	args = append(args, string(headOID))
```